### PR TITLE
Add a `fake.factory` function that generates whole objects

### PR DESCRIPTION
**Pull request by demands**
For example:

```js
type Recipe = {|
  +_id: string,
  +name: string,
  +photo: string,
|};

const recipeCreator = fake.factory({
  _id: fake.objectId,
  name: fake.string,
  photo: fake.uri,
});
const recipe: Recipe = recipeCreator({name: 'Sponge Cake'});
// => {
//      _id: '5153d879dcc2d3e27e689de6',
//      name: 'Sponge Cake',
//      photo: 'http://u6.goodeggs.com/h4mko8c/61n6ut5/c3nr9',
//    }
```

The especially cool thing about this is that the flow typing is properly maintained.

### CODE REVIEW

**Comment by demands** [comment]
Oops! I merged this accidentally. Happy to revert if you have objections @dannynelson  :)

File: src/factory/test.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// @flow
+import {describe, it} from 'mocha';
+import {expect} from 'goodeggs-test-helpers';
+
+import fake from '..';
+import factory from '.';
+
+describe('factory', function () {
+  it('works', function () {
+    type Recipe = {|
```
Comment (dannynelson) [inline src/factory/test.js]: Does this work for non-sealed objects too?

File: readme.md [hunk 2]
```diff
@@ -21,6 +21,28 @@ fake.employee.email(); // => 'rylee.mayert@goodeggs.com
 
 <!-- automatically generated documentation starts here. -->
 
+#### `fake.factory`
+
+`any`
+
+Makes a function that can build up arbitrary objects. For example:
+```js
+const recipeCreator = fake.factory({
```
Comment (dannynelson) [inline readme.md]: Technically, it's a real factory, not a fake factory 😉

File: src/factory/index.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+type DefaultCreatorsFor<T: Object> = $ObjMap<T, <V>(V) => (() => V)>;
```
Comment (ndhoule) [inline src/factory/index.js]: Are you intentionally omitting the `@flow` pragma here?

Comment (demands) [reply src/factory/index.js]: Ooooops.

Comment (demands) [reply src/factory/index.js]: Fixed in master. 🙊

**Pull request merged**